src/StaticAnalysis/ExecutableLinesFindingVisitor.php

```diff
@@ -172,9 +172,40 @@ private function getLines(Node $node): array
                 return [$node->getEndLine()];
             }
 
-            if ($node->items[0] instanceof ArrayItem) {
-                return [$node->items[0]->getStartLine()];
+            $lines = [];
+
+            foreach ($node->items as $index => $item) {
+                if (!$item instanceof ArrayItem) {
+                    continue;
+                }
+
+                if ($index === 0) {
+                    $lines[] = $item->getStartLine();
+
+                    continue;
+                }
+
+                if ($item->key instanceof BinaryOp ||
+                    $item->value instanceof BinaryOp) {
+                    continue;
+                }
+
+                if ($item->key instanceof Node\Expr\Variable ||
+                    $item->key instanceof Node\Expr\ClassConstFetch
```

Comment:
I would say this will return the unstable behaviour described in https://github.com/sebastianbergmann/php-code-coverage/issues/889